apps/local_laplacian/local_laplacian_generator.cpp

```diff
@@ -124,17 +117,16 @@ class LocalLaplacian : public Halide::Generator<LocalLaplacian> {
             }
             outGPyramid[0]
                 .compute_at(output, y).vectorize(x, 8);
-            for (int j = 5; j < J; j++) {
+            for (int j = 5; j < pyramid_levels; j++) {
                 inGPyramid[j].compute_root();
                 gPyramid[j].compute_root().parallel(k);
                 outGPyramid[j].compute_root();
             }
         }
-
-        return output;
     }
 private:
     Var x, y, c, k;
+    Func remap, gray, inGPyramid[maxJ], gPyramid[maxJ], outGPyramid[maxJ];
```

Comment:
I definitely like this. It's worth working a few examples, of course, to further develop it. The obvious cases seem to be the camera pipe, and also something where a module would be scheduled differently in different contexts which would historically have been guarded with C++-level `if`s (vectorizing an internal stage within a module which is `compute_at` some level of the module's output, in cases where the module might be inlined into its uses). This seems to do a nice job with the scoping challenge in things like camera pipe, but it's less clear to me if it has an impact on things that need to propagate through downstream uses in the face of inlining, etc.